spark/v4.0/spark-extensions/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckViews.scala

```diff
@@ -105,7 +106,7 @@ object CheckViews extends (LogicalPlan => Unit) {
   ): Unit = {
     val newCyclePath = cyclePath :+ currentViewIdent
     if (currentViewIdent == viewIdent) {
-      throw new AnalysisException(String.format("Recursive cycle in view detected: %s (cycle: %s)",
+      throw new IcebergAnalysisException(String.format("Recursive cycle in view detected: %s (cycle: %s)",
```

Comment:
Spark 4.0 doesn't allow constructing an AnalysisException with error message only. We have to provide an error class. In order to get around this, I have added `IcebergAnalysisException`, which only needs an error message.